nuxeo-retention-core/src/main/java/org/nuxeo/ecm/retention/listener/RetentionRecordCheckerListener.java

```diff
@@ -38,25 +42,61 @@
 
     @Override
     public void handleEvent(EventBundle events) {
-        for (Event event : events) {
-            checkRecord(event);
+        Map<String, List<String>> docsToCheckAndEvents = new HashMap<String, List<String>>();
+        String ignoreId = null;
+        if (events.containsEventName(RetentionService.RETENTION_CHECKER_LISTENER_IGNORE_EVENT)) {
```

Comment:
I know, it was just to prevent from having a "if hell" by removing the ignored document before looping.